backend/types/express/index.d.ts

```diff
@@ -0,0 +1,14 @@
+import { IUser } from '../models/user';
+
+declare global {
+  namespace Express {
+    interface Request {
+      user?: {
```

Comment:
Is it truly required ? Because here we assume the default Request type if of the user type right? I prefer using User type instead of Request type , because request can be multiple we can gloablize it to user